server/routes/user.py

```diff
@@ -47,23 +48,48 @@ def oauth_login():
             user = User(oauth_id=sub, name=name, email=email)
             db.session.add(user)
             db.session.commit()
-            transaction = Transaction(
-                user_id=user.id, ticket_amount=1000, activity="Sign up bonus"
-            )
-            db.session.add(transaction)
-            db.session.commit()
+            session["user_id"] = user.id
+            return {"success": True, "new_user": True}
 
         session["user_id"] = user.id
 
-        return {"success": True, "user_id": session["user_id"]}
+        if user.username is None:
+            return {"success": True, "new_user": True}
+
+        return {"success": True, "new_user": False, "user_id": session["user_id"]}
 
     except json.decoder.JSONDecodeError:
         return {"error": "Malformed request"}, 400
 
-
 def check_username(username):
     return User.query.filter_by(username=username).scalar() is not None
 
+@user_bp.route("/api/login/newuser", methods=["POST"])
+def oauth_newuser():
+    try:
+        data = json.loads(request.data)
+        username = data["user"]
+
+        if "user_id" not in session:
```

Comment:
use server.routes.decorators.login_required

Look up `@login_required` in the codebase

Reply:
Okay, fixed it.